x-pack/plugins/maps/public/classes/sources/tms_source/index.ts

```diff
@@ -5,8 +5,9 @@
  * 2.0.
  */
 
+import { DataFilters } from '../../../../common/descriptor_types';
 import { ISource } from '../source';
 
 export interface ITMSSource extends ISource {
-  getUrlTemplate(): Promise<string>;
+  getUrlTemplate(dataFilters?: DataFilters): Promise<string>;
```

Comment:
Make `getUrlTemplate(dataFilters: DataFilters)`. dataFilters is always provided by the caller.